clients/src/test/java/org/apache/kafka/clients/consumer/internals/ShareHeartbeatRequestManagerTest.java

```diff
@@ -422,6 +425,29 @@ public void testHeartbeatResponseOnErrorHandling(final Errors error, final boole
         }
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {SHARE_PROTOCOL_NOT_SUPPORTED_MSG})
+    public void testUnsupportedVersion(String errorMsg) {
```

Comment:
this is truly `testUnsupportedVersionGeneratedOnTheClient` (client response with exception). Renaming will maybe clarify, and we should ensure we have coverabe for something like `testUnsupportedVersionGeneratedOnTheBroker` (client response with `ErrorCode`), is it already covered?

The outcome of both tests should be the same, but the execute diff code paths modified in this PR (handleSpecificError and handleSpecificFailure/exception)